sdk/core/azure_core_test/src/recorded.rs

```diff
@@ -18,7 +18,14 @@ static TEST_PROXY: OnceCell<Result<Arc<Proxy>>> = OnceCell::const_new();
 ///
 /// The [Test Proxy](https://github.com/Azure/azure-sdk-tools/blob/main/tools/test-proxy/Azure.Sdk.Tools.TestProxy/README.md) service will be started as needed.
 /// Every `#[recorded::test]` will call this automatically, but it can also be called manually by any other test e.g., those attributed with `#[tokio::test]`.
-pub async fn start(ctx: &TestContext, options: Option<ProxyOptions>) -> Result<Session> {
+///
+/// This function will return `Ok(None)` if the test is running in live mode and should not use the test proxy at all.
+pub async fn start(ctx: &TestContext, options: Option<ProxyOptions>) -> Result<Option<Session>> {
+    // Live tests don't use test-proxy.
+    if ctx.test_mode() == azure_core::test::TestMode::Live {
+        return Ok(None);
+    }
```

Comment:
Then why declare a `TestContext` parameter at all?

Reply:
Oh, I see. You actually use it. But then there's no need to call `start()`. I guess I'm puzzled as to why this is needed. If we truly need it, fine; my questions are merely how we got to this point because it may alter my assertions about how this is used.

Reply:
Never mind. We talked offline and this makes sense. No callers needed to change because, while I capture it in the macro, I don't use it yet (but will). For live tests that want to use it or when `AZURE_TEST_MODE=live`, we'll need some way of signaling that test-proxy wasn't started unnecessarily, so this will no doubt be useful.